lib/gpfpiso.gi

```diff
@@ -888,5 +964,992 @@ function( G, series )
+
+    Add(rels,(gens[1]*gens[3])^2);
+    Add(rels,(gens[1]*gens[4])^2);
+    Add(rels,(gens[2]*gens[4])^2);
+    Add(rels,(gens[1]*gens[5])^2);
+    Add(rels,(gens[2]*gens[5])^2);
+    Add(rels,(gens[4]*gens[5])^2);
+    Add(rels,(gens[1]*gens[6])^2);
+    Add(rels,(gens[2]*gens[6])^2);
+    Add(rels,(gens[3]*gens[6])^2);
+    Add(rels,(gens[4]*gens[6])^2);
+    if n>=7 then
+      Add(rels,(gens[6]*gens[7])^3);
+      for i in [1..5] do
+        Add(rels,(gens[i]*gens[7])^2);
+      od;
+    fi;
+    if n=8 then
+      Add(rels,(gens[7]*gens[8])^3);
+      for i in [1..6] do
+        Add(rels,(gens[i]*gens[7])^2);
+      od;
+    elif n>8 then 
+      Error("E>8 does not exist");
+    fi;
+  elif ser="F" and n=4 then
+    Add(rels,(gens[1]*gens[2])^3);
+    Add(rels,(gens[2]*gens[3])^4);
+    Add(rels,(gens[3]*gens[4])^3);
+
+    Add(rels,(gens[1]*gens[3])^2);
+    Add(rels,(gens[1]*gens[4])^2);
+    Add(rels,(gens[2]*gens[4])^2);
+  elif ser="G" and n=2 then
+    Add(rels,(gens[1]*gens[2])^6);
+  else
+    Error("series ",ser," not yet done");
+  fi;
+  return f/rels;
+end);
+
+BindGlobal("IsomorphismFpGroupForWeyl",
+function(G)
+local iso,n,fn,sz,bigcount,tryweyl;
+
+  tryweyl:=function(ser,n)
+  local H,isp,P,iso;
+    H:=WeylGroupFp(ser,n);
+    isp:=IsomorphismPermGroup(H);
+    P:=Image(isp,H);
+    iso:=IsomorphismGroups(G,P);
+    if iso<>fail then
+      P:=List(GeneratorsOfGroup(H),
+        x->PreImagesRepresentative(iso,ImagesRepresentative(isp,x)));
+      iso:=GroupHomomorphismByImagesNC(G,H,P,GeneratorsOfGroup(H));
+    fi;
+    return iso;
+  end;
+
+  # try to identify Weyl series
+  n:=0;
+  repeat
+    n:=n+1;
+    fn:=Factorial(n);
+    bigcount:=0;
+    sz:=fn*(n+1);
+    if Size(G)<sz then bigcount:=bigcount+1;
+    elif Size(G)=sz and IsSymmetricGroup(G) then
+      # try An
+      if HasIsomorphismFpGroup(G) then
+        G:=Group(GeneratorsOfGroup(G));
+        SetSize(G,sz);
+        IsSymmetricGroup(G);
+      fi;
+      return IsomorphismFpGroup(G); # uses symmetric method
+    fi;
+    sz:=2^n*fn;
+    if Size(G)<sz then bigcount:=bigcount+1;
+    elif Size(G)=sz then
+      iso:=tryweyl("B",n);
+      if iso<>fail then return iso;fi;
+    fi;
+    sz:=2^(n-1)*fn;
+    if Size(G)<sz then bigcount:=bigcount+1;
+    elif n>=4 and Size(G)=sz then
+      iso:=tryweyl("D",n);
+      if iso<>fail then return iso;fi;
+    fi;
+  until bigcount=3;
+
+  if Size(G)=12 then
+    iso:=tryweyl("G",2);
+    if iso<>fail then return iso;fi;
+  elif Size(G)=1152 then
+    iso:=tryweyl("F",4);
+    if iso<>fail then return iso;fi;
+  elif Size(G)=51840 then
+    iso:=tryweyl("E",6);
+    if iso<>fail then return iso;fi;
+  elif Size(G)=72*Factorial(8) then
+    iso:=tryweyl("E",7);
+    if iso<>fail then return iso;fi;
+  elif Size(G)=192*Factorial(10) then
+    iso:=tryweyl("E",8);
+    if iso<>fail then return iso;fi;
+  fi;
+
+  return fail;
+end);
+
+# rewriting systems for simple groups based on BN pairs, following Schmidt
```

Comment:
Can you give a more explicit reference than just "following Schmidt"? Like, is there a paper or a thesis, or something?

I may be missing something, but, aren't these "standard" ideas when working with BN-groups, and present in e.g. Cohen-Murray-Taylor?

Reply:
It is basically Schmidt's only paper (and his thesis). But I see the point that there might be another person names Schmidt.
Added a referecne.